dbms/src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -3277,7 +3277,7 @@ void StorageReplicatedMergeTree::sendRequestToLeaderReplica(const ASTPtr & query
         leader_address.host,
```

Comment:
So generally it will use the same credential to connect to leader replica which were used be current user?

Also: change obsolete comment few lines above.

Reply:
can be done like interserver_http_port but in that case password should be stored in zookeeper as open text. Some options of encrypting that password in zookeeper are possible (for example with common key for whole cluster), but can be not clear in configuration and can't be considered 100% safe.

Looking in cluster config (which can contain user / password) - cluster config is not required for ReplicatedMergeTree inself. 

The best would be to have some other authentication protocol for interserver communication, which will not require to know the passwords of each other, but it's a big change.

I was thinking about storing user and passwords for interserver communication in separate config section, which can be shared between cluster (Distributed) and ReplicatedMergeTree, something like 
```xml
<interserver_credentials>
  <server1>
     <user>abc</user>
     <password>abc</password>
   </server1>
   <server2>
     <user>abc</user>
     <password>abc</password>
   </server2>
</interserver_credentials>
```

In that case it should be quite easy to lookup for credentials both for cluster config and for ReplicatedMergeTree.